python/housinginsights/ingestion/LoadData.py

```diff
@@ -286,8 +289,267 @@ def rebuild(self):
+        with key "items", which contains a list of dictionaries each with 'grouping'
+        and 'count'.
+
+        Returns items as dictionary with group and division result as
+        key/value pairs instead of list of dictionaries.
+        """
+        items = {}
+        if numerator_data['items'] is None:
+            items = None
+        else:
+            for n in numerator_data['items']:
+                # TODO what should we do when a matching item isn't found?
+                matching_d = next((item for item in denominator_data['items'] if
+                                   item['group'] == n['group']),
+                                  {'group': '_unknown', 'value': None})
+                if matching_d['value'] is None or n['value'] is None:
+                    divided = None
+                else:
+                    divided = n['value'] / matching_d['value']
+
+                # item = dict({'group': n['group'],
+                #              'value': divided})
+                items[n['group']] = divided
+
+        return {'items': items, 'grouping': numerator_data['grouping'],
+                'data_id': numerator_data['data_id']}
+
+    def _census_with_weighting(self, data_id, grouping):
+        """
+        Zone facts table helper function to get data from our census table.\
+
+        :param data_id: must either be a column name or a custom calculated
+        value.
+
+        :param grouping: either 'census_tract', 'ward',
+        or 'neighborhood_cluster'.
+
+        :return: dictionary with 'items', 'grouping', and 'data_id' as keys.
+        Items stores the results for the calculated data_id for the grouping.
+        """
+
+        calculated_values = {
+            'poverty_rate': 'population_poverty',
+            'fraction_black': 'population_black',
+            'income_per_capita': 'aggregate_income',
+            'labor_participation': 'population_working',
+            'fraction_foreign': 'population_foreign',
+            'fraction_single_mothers': 'population_single_mother'
+        }
+
+        if data_id in calculated_values:
+
+            field = calculated_values[data_id]
+            numerator = self._get_weighted_census_results(grouping, field)
+            denominator = self._get_weighted_census_results(grouping,
+                                                            'population')
+
+            api_results = self._items_divide(numerator, denominator)
+            api_results['data_id'] = data_id
+
+        # If data_id isn't one of our custom ones, assume it is a column name
+        else:
+            api_results = self._get_weighted_census_results(grouping, data_id)
+
+            # transform items to dict with group/value instead of list of dict
+            temp_items = {}
+            for item in api_results['items']:
+                temp_items[item['group']] = item['value']
+
+            api_results['items'] = temp_items
```

Comment:
As noted above, don't need to store this on the 'items' field anymore - that was just relevant for how the API needed to serve its results in json format